File: lib/rgfa/line/header/field_data.rb
# Disallow editing the VN tag in connected header lines
# @tested_in api_header
module RGFA::Line::Header::FieldData

  private

  def set_existing_field(fieldname, value, set_reference: false)
    if fieldname == :VN and !get(:VN).nil? and self.connected?
      raise RGFA::RuntimeError,
        "The value of the header tag VN cannot be edited\n"+
        "For version conversion use to_gfa1 or to_gfa2"
    else
      super
    end
  end

end

File: gfapy/line/header/field_data.py
import gfapy

class FieldData:
  """
  Disallow editing the VN tag in connected header lines
  """

  def _set_existing_field(self, fieldname, value, set_reference=False):
    if fieldname == "VN" and self.get("VN") is not None and self.is_connected():
      raise gfapy.RuntimeError(
        "The value of the header tag VN cannot be edited\n"+
        "For version conversion use to_gfa1 or to_gfa2")
    else:
      super()._set_existing_field(fieldname, value,
                                  set_reference=set_reference)
